homeassistant/components/reolink/host.py

```diff
@@ -363,32 +365,54 @@ async def handle_webhook(
         self, hass: HomeAssistant, webhook_id: str, request: Request
     ):
         """Shield the incoming webhook callback from cancellation."""
-        await asyncio.shield(self.handle_webhook_shielded(hass, webhook_id, request))
+        self._webhook_read_done.clear()
+        shielded_task = asyncio.create_task(
+            self._handle_webhook(hass, webhook_id, request)
+        )
+        asyncio.shield(shielded_task)
+        # To protect against garbage collection
+        self._webhook_tasks.add(shielded_task)
+        shielded_task.add_done_callback(self._webhook_tasks.discard)
```

Comment:
This is the exact logic that we have in `self.hass.async_create_task` which should be used.

Reply:
Done, thanks for pointing this out, that simplifies things a bit 👍
Do you have any other suggestions?